dipy/tracking/tests/test_utils.py

```diff
@@ -37,6 +37,60 @@ def make_streamlines():
     return streamlines
 
 
+def test_cci():
+    # two identical streamlines should raise an error
+    mysl = np.array([np.arange(10)] * 3).T
+    test_streamlines = list([mysl])*2
```

Comment:
Due to memory issue, you should avoid having a list of `np.array` as much as possible. This is for this reason, we created `Streamlines` object. Can you replace `list` with `Streamlines`?

Reply:
Thanks, @skoudoro.

Something I noticed while fixing this: you get an error if the type isn't double (my toy streamlines in tests were initially integers)... I just fixed it in the test to make the toys floats. Can we assume that inputs to set_number_of_points will always be floats or is this an issue?

dipy.tracking.streamlinespeed.set_number_of_points()
ValueError: Buffer dtype mismatch, expected 'double' but got 'long'

Reply:
Indeed, I think this is an issue. But I would like to have @Garyfallidis opinion.